mafContrib/transientAsciiSEDMetric.py

```diff
@@ -0,0 +1,662 @@
+            flt_times = list(time[np.where(filters == flt)[0]])
+            # Initialize the band magnitude list.
+            filter_mag = []
+            # Get all band magnitudes. Currently must be in loop, method fails
+            # in the scipy interpolation if attempting to vectorize according
+            # to sncosmo documentation.
+            for obs_time in flt_times:
+                filter_mag.append(
+                    redshifted_model.bandmag("lsst" + flt, "ab", obs_time)
+                )
+            # Set light_curve_mags for array indices corresponding to observations of the
+            # current filter.
+            light_curve_mags[np.where(filters == flt)[0]] = np.array(filter_mag)
+            self.light_curve_mags = light_curve_mags
+
+    def evaluate_all_detection_criteria(self, dataSlice):
+        """
+        Wrapper function to setup loop for each transient light curve and
+        evaluate all detection criteria.
+
+        Parameters:
+        -----------
+            dataSlice : numpy.array
+                Numpy structured array containing the data related to the visits
+                provided by the slicer.
+
+        Returns (to self):
+        ------------------
+            transient_detected: np.array, boolean
+                Array containing boolean tracking variable whether transient is
+                detected by passing all criteria.
+            num_detected: int
+                Scalar value of the number of transients that were detected in
+                total between all phase shifts considerd.
+
+        """
+        # Track whether each individual light curve was detected.
+        # Start with the assumption that it is True, and if it fails
+        # criteria then becomes False.
+        self.transient_detected = np.ones(len(np.unique(self.transient_id)), dtype=bool)
+
+        # Loop through each lightcurve and check if it meets requirements.
+        for i, (start_ind, end_ind) in enumerate(
+            zip(self.transient_start_index, self.transient_end_index)
+        ):
+            t_id = i
+            while self.transient_detected[t_id] == True:
+                # If there were no observations at all for this lightcurve:
+                if start_ind == end_ind:
+                    self.transient_detected[t_id] = False
+
+                self.observation_epoch_above_thresh = self.observation_epoch[
+                    start_ind:end_ind
+                ][np.where(self.obs_above_SNR_threshold[start_ind:end_ind])]
+
+                self.evaluate_pre_time_detection_criteria(t_id)
+                self.evaluate_phase_section_detection_criteria(t_id)
+                self.evaluate_number_filters_detection_criteria(
+                    dataSlice, start_ind, end_ind, t_id
+                )
+                self.evaluate_filter_in_time_detection_criteria(t_id)
+                # Finished with current set of conditions
+
```

Comment:
I am a little unsure if this while loop is functioning like you might be hoping? 
Was the idea to test simple criteria first, and if they fail, then to stop testing the individual lightcurve? Or to test all criteria for each lightcurve (which is what I believe this while loop is doing)? If that was the intent, this is fine -- but I think you could drop the while loop entirely in that case?

Example:
```>>> x = True
>>> while x == True:
...     print('first criteria')
...     x = False
...     print('second criteria')
...     x = False
...     print('third criteria')
...     x = False
... 
first criteria
second criteria
third criteria
```

Reply:
Thank you for point this is out! Indeed, as we start with the assumption that the transient _is_ detected, if any criterion fails then it should continue with the next transient. I misunderstood that the while loop would be broken if `transient_detected[t_id]` was set to False anywhere within the loop. I have fixed this.